### PR TITLE
MDEV-16010 Too many rows with AS OF point_in_the_past_or_NULL

### PR DESCRIPTION
**Pull request by midenok**
Depends on #727.

This code is submitted under the BSD-new license.

### CODE REVIEW

File: sql/table.cc [hunk 1]
```diff
@@ -8716,7 +8716,15 @@ bool TR_table::query_sees(bool &result, ulonglong trx_id1, ulonglong trx_id0,
 
   if (trx_id1 == ULONGLONG_MAX || trx_id0 == 0)
   {
-    result= true;
+    if (trx_id0 != trx_id1)
```
Comment (vuvova) [inline sql/table.cc]: eh, isn't that guaranteed? if trx_id0 == trx_id1 you return just few lines above.

File: sql/table.cc [hunk 2]
```diff
@@ -8716,7 +8716,15 @@ bool TR_table::query_sees(bool &result, ulonglong trx_id1, ulonglong trx_id0,
...
+      result= true;
+    return false;
+  }
+
+  if (trx_id0 == ULONGLONG_MAX || trx_id1 == 0)
+  {
+    if (trx_id0 != trx_id1)
```
Comment (vuvova) [inline sql/table.cc]: same

File: sql/table.cc (hunk 1 above)
Comment (midenok) [reply sql/table.cc]: No, it's not guaranteed. It's user-defined.

Comment (midenok) [reply sql/table.cc]: Sure. Fixed.

**Comment by vuvova** [comment]
merged into bb-10.3-serg

**Pull request closed without merging**